### ISSUE TITLE
csharp benchmark failures

### ISSUE THREAD

**Issue by apolcyn** [grpc/grpc#9587]
https://grpc-testing.appspot.com/job/gRPC_performance_pull_requests/9029/

```
benchmarks.qps_json_driver.csharp_protobuf_async_unary_qps_unconstrained
benchmarks.qps_json_driver.csharp_protobuf_async_unary_ping_pong
benchmarks.qps_json_driver.csharp_protobuf_sync_to_async_unary_ping_pong
benchmarks.qps_json_driver.csharp_protobuf_async_streaming_ping_pong
```

all have the same error: 
```
E0203 23:24:21.810027357   29558 driver.cc:444]              Client 0 had an error Connect Failed
```

<s>Possibly fixed by https://github.com/grpc/grpc/pull/9580</s>

**Comment by apolcyn** [comment]
also seen in https://grpc-testing.appspot.com/job/gRPC_performance_pull_requests/9045/#showFailuresLink

or #9548

**Comment by apolcyn** [comment]
seen again https://grpc-testing.appspot.com/job/gRPC_performance_pull_requests/9046/

**Comment by apolcyn** [comment]
fixed by #9612

**Comment by jtattermusch** [comment]
See also https://github.com/grpc/grpc/issues/9499.
